### ISSUE TITLE
Make device properties prettier

### ISSUE THREAD

**Issue by leofang** [NVIDIA/cuda-python#210]
Similar to #205 but for `Device` properties. Currently we return all of them at once, with the C enumerator names as the keys
https://github.com/NVIDIA/cuda-python/blob/213baf40966d8c449c62cce2c55e945a99c452b1/cuda_core/cuda/core/experimental/_device.py#L82-L85
We should perhaps make the returned object a Python class instead of dict, with nice, pythonic attributes.

**Comment by leofang** [comment]
xref: our C++ counterpart (`cudax`) https://github.com/NVIDIA/cccl/pull/2084

**Comment by ksimpson-work** [comment]
Piotr suggested we base the exposure on cudaDeviceAttribtues rather than cudaDeviceProperties to reduce the minor version compatibility considerations. WDYT @leofang

**Comment by leofang** [comment]
Inside you `DeviceProperties` class we could consider using that. But I don't how does this help with
> to reduce the minor version compatibility considerations

**Comment by leofang** [comment]
Yeah followed-up internally, the device prop struct is frozen by major.0, so any new attributes added in major.minor will not be exposed to the prop struct until the next major release, whereas the attribute query API can immediately gain support by the minor release that adds the new attribute. Let's move away from the device prop struct, and query & cache the attributes lazily (on the as-needed basis).
